core/eolearn/core/extra/ray.py

```diff
@@ -60,15 +60,61 @@ def _ray_workflow_executor(workflow_args: _ProcessingData) -> WorkflowResults:
     return RayExecutor._execute_workflow(workflow_args)
 
 
-def _progress_bar_iterator(futures: list, update_interval: float = 0.5) -> Generator:
-    """A utility to help track finished ray processes
+def parallelize_with_ray(
+    function: Callable[[_InputType], _OutputType], *params: Iterable[_InputType], **tqdm_kwargs: Any
+) -> List[_OutputType]:
+    """Parallelizes function execution with Ray.
 
-    Note that using tqdm(futures) directly would cause memory problems and is not accurate
+    Note that this function will automatically connect to a Ray cluster, if a connection wouldn't exist yet. But it
+    won't automatically shut down the connection.
 
-    :param futures: List of `ray` futures.
-    :param update_interval: How many seconds to wait before updating progress bar.
-    :return: A `None` value generator that accurately shows progress of futures.
+    :param function: A normal function that is not yet decorated by `ray.remote`.
+    :param params: Iterables of parameters that will be used with given function.
+    :param tqdm_kwargs: Keyword arguments that will be propagated to `tqdm` progress bar.
+    :return: A list of results in the order that corresponds with the order of the given input `params`.
     """
-    while futures:
-        done, futures = ray.wait(futures, num_returns=len(futures), timeout=float(update_interval))
-        yield from (None for _ in done)
+    ray.init(address="auto", ignore_reinit_error=True)
```

Comment:
Its either this or checking `ray.is_initialized()` which we also do in `RayExecutor`. I'm not sure what is better, although it would be good to be consistent. Advantage with `ray.init` is that you might not need to call it outside this function. But I don't know if ever this can become a problem.

Reply:
I think it's best if the user initializes ray themselves. Maybe they'd want to connect to a certain address or something like that.

Reply:
Done + added a test for this.